### PR TITLE
Followup tweaks to #413

### PR DESCRIPTION
**Pull request by DavisVaughan**
- Improved comment
- Use `exec()` rather than `expr()` + `eval_bare()`

### CODE REVIEW

File: R/compare.R [hunk 1]
```diff
@@ -142,16 +142,12 @@ order_proxy <- function(proxy, direction = "asc", na_value = "largest") {
...
     if (vec_size(proxy) == 0L) {
       return(integer(0L))
     }
-
     args <- unname(proxy)
-    order_expr <- expr(
-      base::order(!!!args, decreasing = decreasing, na.last = na.last)
-    )
-    eval_bare(order_expr)
+    exec(base::order, !!!args, decreasing = decreasing, na.last = na.last)
```
Comment (lionel-) [inline R/compare.R]: Maybe `"order"` to improve backtraces in case `order()` fails?

Comment (DavisVaughan) [reply R/compare.R]: Is the backtrace also going to be ugly since `!!!args` is being done?

Comment (lionel-) [reply R/compare.R]: No because these are scalars. Also the function is most important because it helps to identify the source of a problem.

Comment (lionel-) [reply R/compare.R]: (oops sorry these are not scalars)

**Pull request merged**